### ISSUE TITLE
possible self breach case 

### ISSUE THREAD

**Issue by LNBIG-COM** [lightningnetwork/lnd#2741]
### Your environment

* version of `lnd`: 0.5.2-beta commit=v0.5.2-beta
* which operating system (`uname -a` on *Nix): CentOS 6
* version of `btcd`, `bitcoind`, or other backend 0.17.1

The user DEREC contacted in Slack channel with me. After a careful study of what he wrote, I was horrified, to be honest. My node LND-21 broadcasted old state of channel and his node broadcasted breach transaction. After the funds (2M satoshies) was sent to his node. After my investigation i found that the node was not stopped and not restarted in that day. So old state of channel was broadcated by LND by not-catched bug i think. And after investigation of logs i didn't found some warnings about this case! In my side this to be viewed as force closing of channel!

Please help to me: what symptomps in my logs may indicate such situations?

I attached his state of channel, transactions and my logs:

From his side:

    Then :lncli closedchannels
    {
       "channels": [
           {
               "channel_point": "c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0",
               "chan_id": "618156432394223616",
               "chain_hash": "000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f",
               "closing_tx_hash": "35008a2b6182431ca635960349a6901b98891b5ef19450753569fe56e7ed1dfd",
               "remote_pubkey": "02c91d6aa51aa940608b497b6beebcb1aec05be3c47704b682b3889424679ca490",
               "capacity": "2000000",
               "close_height": 565348,
               "settled_balance": "45",
               "time_locked_balance": "0",
               "close_type": "BREACH_CLOSE"

From my side:

    2019-02-08 21:11:14.757 [INF] FNDG: Target commit tx sat/kw for pendingID(1d4aa3bb6086ffc296c057bdf255d93356ec2d4ff3820eb5874741300841dda3): 3769
    2019-02-08 21:11:14.757 [INF] FNDG: Starting funding workflow with 83.213.102.133:9735 for pendingID(1d4aa3bb6086ffc296c057bdf255d93356ec2d4ff3820eb5874741300841dda3)
    2019-02-08 21:11:18.004 [INF] FNDG: Recv'd fundingResponse for pendingID(1d4aa3bb6086ffc296c057bdf255d93356ec2d4ff3820eb5874741300841dda3)
    2019-02-08 21:11:18.007 [INF] FNDG: pendingChan(1d4aa3bb6086ffc296c057bdf255d93356ec2d4ff3820eb5874741300841dda3): remote party proposes num_confs=3, csv_delay=240
    2019-02-08 21:11:18.007 [INF] FNDG: Generated ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0) for pendingID(1d4aa3bb6086ffc296c057bdf255d93356ec2d4ff3820eb5874741300841dda3)
    2019-02-08 21:11:18.184 [INF] FNDG: Broadcasting funding tx for ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): (*wire.MsgTx)(0xc0580f14c0)({
    2019-02-08 21:11:18.184 [INF] LNWL: Inserting unconfirmed transaction c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d
    2019-02-08 21:11:18.190 [INF] CNCT: Creating new ChannelArbitrator for ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0)
    2019-02-08 21:11:18.191 [INF] CNCT: ChannelArbitrator(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): starting state=StateDefault
    2019-02-08 21:11:18.192 [INF] NTFN: New spend subscription: spend_id=244, outpoint=c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0, height_hint=562191
    2019-02-08 21:11:18.206 [INF] FNDG: Finalizing pendingID(1d4aa3bb6086ffc296c057bdf255d93356ec2d4ff3820eb5874741300841dda3) over ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0), waiting for channel open on-chain
    2019-02-08 21:11:18.206 [INF] CNCT: Close observer for ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0) active
    2019-02-08 21:11:18.206 [INF] NTFN: New confirmation subscription: txid=c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d, numconfs=3
    2019-02-08 21:11:18.206 [INF] FNDG: Waiting for funding tx (c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d) to reach 3 confirmations
       PreviousOutPoint: (wire.OutPoint) c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:1,
    2019-02-08 23:25:45.627 [INF] LNWL: Marking unconfirmed transaction c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d mined in block 562210
    2019-02-08 23:40:48.222 [INF] NTFN: Dispatching 3 conf notification for c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d
    2019-02-08 23:40:48.222 [INF] FNDG: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0) is now active: ChannelID(3db9473242cbf83e2e4efc2323aa614e71d1fbcfff70680afaf0ef4ed3691dc6)
    2019-02-08 23:40:52.455 [INF] CRTR: New channel discovered! Link connects 02c91d6aa51aa940608b497b6beebcb1aec05be3c47704b682b3889424679ca490 and 033c87f6e5e202a4569d6d074da09cad4210a25bb89f00d6f53caa7429f6e55eed with ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): chan_id=61815643239
    2019-02-08 23:40:52.551 [INF] NTFN: New confirmation subscription: txid=c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d, numconfs=6
    2019-02-08 23:40:52.780 [INF] PEER: New channel active ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0) with NodeKey(033c87f6e5e202a4569d6d074da09cad4210a25bb89f00d6f53caa7429f6e55eed)
    2019-02-08 23:40:52.782 [INF] HSWC: ChannelLink(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0) is starting
    2019-02-08 23:40:52.782 [INF] CNCT: Attempting to update ContractSignals for ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0)
    2019-02-08 23:40:52.782 [INF] HSWC: HTLC manager for ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0) started, bandwidth=1977272000 mSAT
    2019-02-09 00:17:13.929 [INF] NTFN: Dispatching 6 conf notification for c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d
    2019-02-09 00:17:13.930 [INF] FNDG: Announcing ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0), short_chan_id=(*lnwire.ShortChannelID)(0xc003bb4180)(562210:2171:0)
    2019-02-09 00:35:23.493 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 3389 sat/kw
    2019-02-09 02:10:53.456 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 2961 sat/kw
    2019-02-09 07:11:48.117 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 829 sat/kw
    2019-02-09 07:59:32.235 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 709 sat/kw
    2019-02-09 08:45:05.157 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 829 sat/kw
    2019-02-09 09:05:35.042 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 583 sat/kw
    2019-02-09 11:34:07.663 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 753 sat/kw
    2019-02-09 12:05:03.168 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 530 sat/kw
    2019-02-09 15:27:37.048 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 2428 sat/kw
    2019-02-09 19:51:50.007 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 1633 sat/kw
    2019-02-09 20:46:09.045 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 2521 sat/kw
    2019-02-10 02:38:22.271 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 1267 sat/kw
    2019-02-10 03:32:12.318 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 2520 sat/kw
    ...
    2019-02-28 14:26:01.693 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 5033 sat/kw
    2019-02-28 16:12:51.912 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 5552 sat/kw
    2019-02-28 21:49:35.690 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 4804 sat/kw
    2019-03-01 15:08:57.240 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 5537 sat/kw
    2019-03-01 15:36:52.679 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 6108 sat/kw
    2019-03-01 16:08:48.058 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 6776 sat/kw
    2019-03-01 21:04:40.731 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 5827 sat/kw
    2019-03-02 05:01:53.530 [INF] HSWC: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): updating commit fee to 5033 sat/kw
    2019-03-02 14:06:45.182 [INF] CNCT: ChannelArbitrator(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): force closing chan
    2019-03-02 14:06:46.639 [INF] HSWC: ChannelLink(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0) is stopping
    2019-03-02 14:06:46.739 [INF] HSWC: ChannelLink(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0) has exited
    2019-03-02 14:06:46.741 [INF] CNCT: Broadcasting force close transaction, ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): (*wire.MsgTx)(0xc0b289b340)({
       PreviousOutPoint: (wire.OutPoint) c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0,
    2019-03-02 14:06:46.774 [INF] CNCT: ChannelArbitrator(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): noop trigger chainTrigger
    2019-03-02 14:16:07.642 [INF] NTFN: Dispatching spend notification for outpoint=c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0 at height=565348
    2019-03-02 14:16:07.643 [WRN] CNCT: Unprompted commitment broadcast for ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0) 
    2019-03-02 14:16:07.643 [INF] CNCT: Unilateral close of ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0) detected
    2019-03-02 14:16:07.644 [INF] CNCT: ChannelArbitrator(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): remote party has closed channel out on-chain
    2019-03-02 14:16:07.664 [INF] CNCT: ChannelArbitrator(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): trigger remoteCloseTrigger,  going to StateContractClosed
    2019-03-02 14:16:07.678 [INF] CNCT: ChannelArbitrator(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): contract resolutions empty, marking channel as fully resolved!
    2019-03-02 14:16:07.691 [INF] CNCT: ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0) has been fully resolved on-chain at height=565348
    2019-03-02 14:16:07.691 [INF] CNCT: Marking ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0) fully resolved

And LND-21 start timelines:

    $ grep 'LightningWallet opened' ~/.lnd/mainnet-lnd-run.log 
    2018-11-16 17:30:57.512 [INF] LTND: LightningWallet opened
    2018-11-16 19:29:04.372 [INF] LTND: LightningWallet opened
    2018-11-17 09:27:06.075 [INF] LTND: LightningWallet opened
    2018-11-17 09:37:18.482 [INF] LTND: LightningWallet opened
    2018-11-26 18:44:52.997 [INF] LTND: LightningWallet opened
    2018-12-07 16:52:51.065 [INF] LTND: LightningWallet opened
    2019-01-14 21:23:09.415 [INF] LTND: LightningWallet opened
    2019-01-21 16:27:58.044 [INF] LTND: LightningWallet opened
    2019-02-08 14:33:04.704 [INF] LTND: LightningWallet opened
    2019-02-13 20:38:02.434 [INF] LTND: LightningWallet opened
    2019-02-16 11:43:25.296 [INF] LTND: LightningWallet opened
    2019-02-19 10:09:26.749 [INF] LTND: LightningWallet opened
    2019-03-04 19:35:50.259 [INF] LTND: LightningWallet opened
    2019-03-04 19:59:08.580 [INF] LTND: LightningWallet opened

And in my side data from closedchannels:

    {
        "channel_point": "c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0",
        "chan_id": "618156432394223616",
        "chain_hash": "000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f",
        "closing_tx_hash": "35008a2b6182431ca635960349a6901b98891b5ef19450753569fe56e7ed1dfd",
        "remote_pubkey": "033c87f6e5e202a4569d6d074da09cad4210a25bb89f00d6f53caa7429f6e55eed",
        "capacity": "2000000",
        "close_height": 565348,
        "settled_balance": "0",
        "time_locked_balance": "0",
        "close_type": "REMOTE_FORCE_CLOSE"
    },

**Comment by LNBIG-COM** [comment]
By recommendation in whitepaper of LN there are rules to delete a previous commitment transactions from memory. Do this do the LND? If the LND has a new commitment transaction the previous transaction should be removed from memory immediately, right?

**Comment by LNBIG-COM** [comment]
Can the phrase "Unprompted commitment broadcast" to be as symptom of breach from my side for remote node? If yes i see many same phrases in logs by grep... Please help to define same situations in log files.

**Comment by Roasbeef** [comment]
What kind of storage do you use on the nodes? Do you modify the on disk storage at all during normal operation of the node?

That log message just means that the remote party did a unilateral close.

**Comment by LNBIG-COM** [comment]
> What kind of storage do you use on the nodes? Do you modify the on disk storage at all during normal operation of the node?

This server has only one SSD disk: 512GB SSD Samsung 840 PRO

I have backups to other server through rsync (only readonly of course) but in that day i am not even logged to server.
I think blaming the disc in this case is the last thing.
 
> That log message just means that the remote party did a unilateral close.

Yes, i see. So because i asked to you how define by logs this breach from my side? In any case, what happened happened. And this is confirmed on both sides.

Addition info:

The next transaction after funding one was 35008a2b6182431ca635960349a6901b98891b5ef19450753569fe56e7ed1dfd. I see this transaction in LND-21 side in `listchaintxns` as:

    {
        "tx_hash": "35008a2b6182431ca635960349a6901b98891b5ef19450753569fe56e7ed1dfd",
        "amount": "0",
        "num_confirmations": 744,
        "block_hash": "00000000000000000018de36a61e553db5c2872d23a57466b1da9667fe9f1dfe",
        "block_height": 565348,
        "time_stamp": "1551532535",
        "total_fees": "0",
        "dest_addresses": [
            "bc1qxnxqjw237pd9th6ztd3tqaaemuh2nrr7e38g4hjdhnd3y9fs9wwq6cp3my"
        ]
    },

There is amount is "0" so i understood this as penalty. The next transaction in blockchain is 65614f140e8845c9b0ce1271f9e44e1cb398574f4cca762e3bec6444c0decf14. I see this transaction in listchaintxns too as:

    {
        "tx_hash": "65614f140e8845c9b0ce1271f9e44e1cb398574f4cca762e3bec6444c0decf14",
        "amount": "0",
        "num_confirmations": 745,
        "block_hash": "0000000000000000001bc3aded57fa2959eb1251c29b9eb10bbb8a6ae2d83169",
        "block_height": 565349,
        "time_stamp": "1551532690",
        "total_fees": "0",
        "dest_addresses": [
            "bc1qtpxaqqft7jrv8jzfh7f3yyc2ecyxjvaf70hejs"
        ]
    },

There is amount is "0" too. The next TX f939caff81816356c31fb71b1db8bdf18d5fc12cf1f24007a3c3e43969c6b799 my LND wallet doesn't see. It's right because it was not controlled by my node. It's full penalty.

**Comment by Roasbeef** [comment]
> I think blaming the disc in this case is the last thing.

I ask as it appears that the on-disk state and the in-memory state diverged. All operations to modify this state are synchronized, and there's a durability assumption in all the writes. From what I gather:
  * We update the on disk state to store commitment n+1
  * The in-memory state is updated to store commitment n+1
  * We try to force close, and read the state from disk, but we get commitment n
  * During the force close, we see that what was confirmed wasn't our on disk state of n+1, but was n, so we assume it was the remote party force closing

**Comment by LNBIG-COM** [comment]
May be this will help

    2019-03-02 14:06:46.639 [INF] HSWC: ChannelLink(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0) is stopping
    2019-03-02 14:06:46.739 [INF] HSWC: ChannelLink(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0) has exited
    2019-03-02 14:06:46.741 [INF] CNCT: Broadcasting force close transaction, ChannelPoint(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): (*wire.MsgTx)(0xc0b289b340)({
     Version: (int32) 2,
     TxIn: ([]*wire.TxIn) (len=1 cap=1) {
      (*wire.TxIn)(0xc092cf8d20)({
       PreviousOutPoint: (wire.OutPoint) c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0,
       SignatureScript: ([]uint8) {
       },
       Witness: (wire.TxWitness) (len=4 cap=4) {
        ([]uint8) <nil>,
        ([]uint8) (len=72 cap=144) {
         00000000  30 45 02 21 00 fc 6a 40  91 95 2e 6e 46 b4 5d 88  |0E.!..j@...nF.].|
         00000010  29 55 01 d8 cf 4c 46 eb  cf 44 ff cd e3 ea 1d 8b  |)U...LF..D......|
         00000020  b0 ef e8 d8 6d 02 20 11  a5 c1 c5 6b 29 12 5b 98  |....m. ....k).[.|
         00000030  0e 60 fe 94 03 86 4c 61  d3 b3 d0 90 20 c4 20 1d  |.`....La.... . .|
         00000040  d6 dd 73 f7 df a8 9e 01                           |..s.....|
        },
        ([]uint8) (len=71 cap=144) {
         00000000  30 44 02 20 3e 13 e1 32  75 ed 07 c3 79 da ab 09  |0D. >..2u...y...|
         00000010  83 22 e2 d8 09 b5 e9 30  ef 18 2a 6f 3d 45 e4 37  |.".....0..*o=E.7|
         00000020  d0 c1 0a df 02 20 49 ad  62 a5 32 96 ae 6d e1 82  |..... I.b.2..m..|
         00000030  e9 74 c8 90 b4 9c 5c 33  5c d1 83 c6 73 86 06 86  |.t....\3\...s...|
         00000040  3a ba b8 60 77 34 01                              |:..`w4.|
        },
        ([]uint8) (len=71 cap=500) {
         00000000  52 21 02 56 74 38 60 e6  8b d6 15 e4 b1 5f 2b c4  |R!.Vt8`......_+.|
         00000010  18 33 9e 50 cb c2 1f e2  7b fc 2c de 47 fd f4 22  |.3.P....{.,.G.."|
         00000020  2c d8 47 21 02 6c 54 c6  51 25 16 6b 47 fa 66 a5  |,.G!.lT.Q%.kG.f.|
         00000030  3a 65 2f 7a a8 6d ad 31  eb 06 41 9a a6 14 ab 0b  |:e/z.m.1..A.....|
         00000040  ac 40 35 81 9d 52 ae                              |.@5..R.|
        }
       },
       Sequence: (uint32) 2161984912
      })
     },
     TxOut: ([]*wire.TxOut) (len=1 cap=1) {
      (*wire.TxOut)(0xc07ac8db60)({
       Value: (int64) 1996301,
       PkScript: ([]uint8) (len=34 cap=34) {
        00000000  00 20 34 cc 09 39 51 f0  5a 55 df 42 5b 62 b0 77  |. 4..9Q.ZU.B[b.w|
        00000010  b9 df 2e a9 8c 7e cc 4e  8a de 4d bc db 12 15 30  |.....~.N..M....0|
        00000020  2b 9c                                             |+.|
       }
      })
     },
     LockTime: (uint32) 542063054
    })
    
    2019-03-02 14:06:46.741 [INF] LNWL: Inserting unconfirmed transaction 35008a2b6182431ca635960349a6901b98891b5ef19450753569fe56e7ed1dfd
    2019-03-02 14:06:46.774 [INF] CNCT: ChannelArbitrator(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): noop trigger chainTrigger

**Comment by LNBIG-COM** [comment]
> * During the force close, we see that what was confirmed wasn't our on disk state of n+1, but was n, so we assume it was the remote party force closing

It happened in anyway from the remote party force closing because:

1. I was not logged to server in that day and i didn't force closing

2. I didn't use any API in that time

So remote side made force closing. May be i had autopilot as active but i am not sure (i think in that time the autopilot was turned off there). But i think in anyway your autopilot doesn't force channels. Therefore, it may be the following: remote side did force closing, my LND saw this and issued the last transaction (above) but this transaction was as previous state commitment transaction for the remote side and the remote side already had signed by my LND the breach transaction for my whoe funds. So in that time the remote side immediately after this had published the breach transaction. Right?

**Comment by Roasbeef** [comment]
The remote side didn't force close, you did. It was closed due to an
expired HTLC likely.

**Comment by LNBIG-COM** [comment]
> The remote side didn't force close, you did. It was closed due to an expired HTLC likely.

I did not consider this option. And how did you determine that it was due an expired HTLC?

Ok, may be due an expired HTLC... And what is next? I checked now the kernel logs - i didn't have any 
problems with disk. The LND was started:

    2019-02-19 10:09:26.749 [INF] LTND: LightningWallet opened
    2019-03-04 19:35:50.259 [INF] LTND: LightningWallet opened

Between these intervals this bug occured: `2019-03-02 14:06:45.182`

If I could identify such situations from log files I would look for them. But I do not know how to do it. Only if I will sum up all satoshies. But i have many pending channels with unexisted transactions (it's other bug i already wrote this) so i will get incorrect sums. I did by manually the sum of all satoshies from opened channels, pending ones and from list of wallet transactions. So i think it was one same case in LND-21. But there is difficult to do it same with other nodes.

Please don't close this ticket. I found the two closed channels with same status BREACH_CLOSE from my side already. I think the bug is there. But he doesn't captured.

**Comment by LNBIG-COM** [comment]
These BREACH_CLOSE channels from my side are:

    {
	"channels": [
	    {
		"channel_point": "b94043dd3a64a7981edf43f22c224388c399109b084ae422347957018d3cede8:1",
		"chan_id": "618243293758619649",
		"chain_hash": "000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f",
		"closing_tx_hash": "857ae48df1e9ae5b13087c72838d4221a1bee258989cba8c8b57e6640952066f",
		"remote_pubkey": "02ad6fb8d693dc1e4569bcedefadf5f72a931ae027dc0f0c544b34c1c6f3b9a02b",
		"capacity": "354936",
		"close_height": 565192,
		"settled_balance": "123",
		"time_locked_balance": "0",
		"close_type": "BREACH_CLOSE"
	    }
	]
    }

And from LND-31:

    $ l closedchannels --breach
    {
	"channels": [
	    {
		"channel_point": "0628bbbd9091d447f0f897a4762e6ac698c44a9a68caf6166e79920b09461593:1",
		"chan_id": "615344981066383361",
		"chain_hash": "000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f",
		"closing_tx_hash": "984f9919fb4a23eea64c20113bdd15468369810f0bece610a312dc9b80bbd78e",
		"remote_pubkey": "02e6a169b0826dfea51ee8bb81f629ec7445b267ef36354f2dd5012e1e8b69088c",
		"capacity": "281197",
		"close_height": 563951,
		"settled_balance": "0",
		"time_locked_balance": "0",
		"close_type": "BREACH_CLOSE"
	    }
	]
    }

**Comment by Roasbeef** [comment]
We know it was an HTLC expiry due to this line: 
```
2019-03-02 14:06:45.182 [INF] CNCT: ChannelArbitrator(c61d69d34eeff0fa0a6870ffcffbd1714e61aa2323fc4e2e3ef8cb423247b93d:0): force closing chan
```

**Comment by cfromknecht** [comment]
@LNBIG-COM would you by chance be able to provide full logs during that start of LND? I'm curious to see what else was going on in the daemon at the time. You can send them on slack if you don't wish to post here

**Comment by LNBIG-COM** [comment]
> @LNBIG-COM would you by chance be able to provide full logs during that start of LND? I'm curious to see what else was going on in the daemon at the time. You can send them on slack if you don't wish to post here

No problem. I will bring. Do you need logs of start before this problem (2019-02-19 10:09:26.749) or after (2019-03-04 19:35:50.259)? And i can send to you to email because the log file can be big. I can bring to here if you need only some first log lines.

**Comment by LNBIG-COM** [comment]
Here all logs from begining up to present time

https://drive.google.com/file/d/1__OF47DEWvurfqW1Aj2pHgAmFfdbyu4-/view?usp=sharing

The file gzipped and encrypted by key B8BD8132 / 5ADFB098 (Olaoluwa Osuntokun <laolu32@gmail.com>). This key you use for release signing.

**Comment by AbelLykens** [comment]
Since you shared encrypted, @LNBIG-COM Are you OK with @Roasbeef sharing relevant parts in this thread?
I would be interested to see them.

**Comment by Roasbeef** [comment]
> I have backups to other server through rsync (only readonly of course) but in that day i am not even logged to server.

I'd recommend you _stop_ doing this, as it's possible that `rsync` will _mutate_ the source file if it's being updated while `rsync` is running.

**Comment by LNBIG-COM** [comment]
> > I have backups to other server through rsync (only readonly of course) but in that day i am not even logged to server.
> 
> I'd recommend you _stop_ doing this, as it's possible that `rsync` will _mutate_ the source file if it's being updated while `rsync` is running.

Sorry, with all due respect to you, you say nonsense.

For example: the machine A is working LND, the machine B is backup server. If i do at B:

    rsync -az bitcoin@A:./.lnd/ ./backup/./lnd/

In this case the rsync in machine B cannot affect files in machine A because the instance of rsync at machine A only read files and states. Everywhere in the world backups work same way (and mirorring).

**Comment by Roasbeef** [comment]
> Sorry, with all due respect to you, you say nonsense.

Meant the target file (to be copied) not the source file.

**Comment by Roasbeef** [comment]
Fixed by #2749.
